File: Cli_Input_Ruby_curses/Cli_Input_Item.rb

class Cli_Input_Item

	INPUT_CMD_ENTER = 1
	INPUT_CMD_TAB = 2
	INPUT_CMD_UP = 3
	INPUT_CMD_DOWN = 4

	def initialize(type, text)
		@type = type
		@text = text
	end

	def Type_Get()
		return @type
	end

	def Text_Get()
		return @text
	end

end

File: Cli_Input_CS_Test/Cli_Input_CS_Test/Cli_Input_Item.cs
using System;

namespace Cli_Input_CS_Test
{
    class Cli_Input_Item
    {
        protected Input_Cmd_Type Type;
        protected String Text;

        public Cli_Input_Item(Input_Cmd_Type type, String text)
        {
            Type = type;
            Text = text;
        }

        public Input_Cmd_Type Type_Get()
        {
            return Type;
        }

        public String Text_Get()
        {
            return Text;
        }
    }
}
